lib/Sema/TypeCheckConstraints.cpp

```diff
@@ -2323,12 +2323,12 @@ Type TypeChecker::typeCheckParameterDefault(Expr *&defaultValue,
     AutoClosureListener listener(fnType);
     return typeCheckExpression(defaultValue, DC,
                                TypeLoc::withoutLoc(fnType->getResult()),
-                               canFail ? CTP_DefaultParameter : CTP_CannotFail,
-                               TypeCheckExprOptions(), &listener);
+                               CTP_DefaultParameter, TypeCheckExprOptions(),
+                               &listener);
   }
 
   return typeCheckExpression(defaultValue, DC, TypeLoc::withoutLoc(paramType),
-                             canFail ? CTP_DefaultParameter : CTP_CannotFail);
```

Comment:
Is it possible to remove `CTP_CannotFail` now?

Reply:
@xedin Unfortunately it's still used in a couple of places in CSApply :/ Thanks for the review!

Reply:
No worries, soon then :)